components/job/EditForm.vue

```diff
@@ -0,0 +1,331 @@
+<script setup lang="ts">
+  import type { AlertType } from 'composables/useAlerts'
+  const props = defineProps({
+    jobUid: {
+      type: String,
+      required: false,
+      default: null,
+    },
+  })
+  // Fetch job data if props are provided
+  const auth = useAuthStore()
+  const { hasAccess, user } = storeToRefs(auth)
+
+  // fetch job data
+  const { data: job } = await useFetch('/api/job', {
+    method: 'GET',
+    query: {
+      jobUID: props.jobUid,
+    },
+  })
+
+  // fetch company data
+  const { data: companies } = await useFetch('/api/company', {
+    method: 'GET',
+  })
+
+  const localePath = useLocalePath()
+  // redirect on non existing job or lacking access
+  if (
+    (props.jobUid && !job.value) ||
+    (hasAccess.value(['company']) &&
+      props.jobUid &&
+      job.value.companyUID !== user.value?.companyUID)
+  )
+    navigateTo(localePath('/jobs/edit'))
+
+  const initialState = {
+    companyUID: null,
+    deadline: null,
+    description: null,
+    jobType: null,
+    location: null,
+    title: null,
+    uid: null,
+  }
+
+  // Set state to job data if data fetched successfully
+  const state = reactive(
+    props.jobUid && job.value
+      ? { ...Object.values(embedKeyIntoObjectValues(job.value))[0] }
+      : { ...initialState }
+  )
+
+  // Alert state
+  const initialAlertState = {
+    show: false,
+    alertRoute: '',
+    type: undefined as AlertType,
+  }
+
+  const alertState = reactive({
+    ...initialAlertState,
+  })
+
+  // Show appropriate success alert after signing up for job
+  const displaySuccessAlert = (alertRoute: string) => {
+    alertState.alertRoute = alertRoute
+    alertState.type = 'success'
+    alertState.show = true
+  }
+
+  // show appropriate error alert after signing up for job
+  const displayErrorAlert = (alertRoute: string) => {
+    alertState.alertRoute = alertRoute
+    alertState.type = 'error'
+    alertState.show = true
+  }
+
+  // Show appropriate error alert for failed API calls
+  const displayErrorAlertFromMessage = (errorMessage: string) => {
+    const content = getAlertContent(errorMessage)
+    alertState.alertRoute = content.alertRoute
+    alertState.type = content.type
+    alertState.show = content.show
+  }
+
+  const form = ref()
+
+  // Boolean control states
+  const editMode = computed(() => !!state.uid || !!state.jobUID)
+
+  // get company select options
+  const companyList = computed(() => {
+    if (!hasAccess.value(['admin', 'company']) || !job) return null
+
+    if (hasAccess.value(['admin'])) {
+      return embedKeyIntoObjectValues(companies.value).map((company) => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const { uid, name, ...remains } = company
+        return {
+          value: uid,
+          title: name,
+        }
+      })
+    }
+
+    if (hasAccess.value(['company'])) {
```

Comment:
When creating a new company and signed in as company user this does not seem to work properly:

![Screenshot 2024-01-11 at 12 19 02](https://github.com/et-dagen/et-dagen/assets/48122613/dc0b574b-1bc9-4ee2-ae88-1759ee4dee6d)

Reply:
Also, when I sign in as a company user and try to edit a job that is owned by that users company I am redirected to /jobs/edit, even though I shouldn't.